HyperPlay-Gaming/hyperplay-desktop-client
# [Tech] Refactor proxy

Refactor proxy server

Fixes a bug with WalletConnect QR code not showing as well

# Test 

To test we should launch 
- the artifact with submodules (from github action below)
- the artifact without submodules (from a local build after deleting `node_modules/@hyperplay/proxy`  `node_modules/@hyperplay/providers`
- dev mode with submodules
- dev mode without submodules

And we should ensure that when submodules are present, the proxy server works

And when submodules are not present, the proxy server does not start but the rest of the app launches and works fine.

We can check the proxy server is working through connecting a wallet and running this curl command https://docs.hyperplay.xyz/developer-docs/api/get-accounts

^ note that I didn't add this as a qase test as I'm not sure we need to regression test this

---

Use the following Checklist if you have changed something on the Backend or Frontend:
